interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -98,8 +98,22 @@ class NodeCompiler(definitions: ProcessDefinition[ObjectWithMethodDef],
           val defaultCtx = contextWithOnlyGlobalVariables.withVariable(VariableConstants.InputVariableName, Unknown, paramName = None)
           NodeCompilationResult(Map.empty, None, defaultCtx, error)
       }
-    case FragmentInputDefinition(_, params, _) =>
-      NodeCompilationResult(Map.empty, None, Valid(contextWithOnlyGlobalVariables.copy(localVariables = params.map(p => p.name -> loadFromParameter(p)).toMap)), Valid(new Source {}))
+    case frag@FragmentInputDefinition(id, params, _) =>
+      definitions.sourceFactories.get(id) match {
+        case Some(definition) =>
+          val parameters = fragmentDefinitionExtractor.extractParametersDefinition(frag).value
+          val variables: Map[String, TypingResult] = parameters.map(a => a.name -> a.typ).toMap
+          val defaultContextTransformation =
```

Comment:
Why default?

Reply:
Just followed similar naming (from a little bit above) but as you noticed it can be more meaningful - changed to simple `validationContext` :+1:

Reply:
Yep, but in the case above, it was a function that was used as a fallback :P